### PR TITLE
149 add calendar service

### LINKED ISSUE #149: Add Calendar Service
**Short description:**
Add the google calendar service.

**Describe the solution you'd like**
First i want to be able to create a meeting in my google calendar, then in the google calendar of the logged user,
and then i want to be able to have an action when it's the time of the new meeting.

### PR DESCRIPTION
**Pull request by GuyomT**
# Purpose

The purpose of this pull request is to add the service Google Calendar.

Fixes #149 

## Added features

* This service has been added to the application. It includes:
  * A reaction which create an event in the calendar

### CODE REVIEW

File: server/services/calendar/calendar_init.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+const createCalendarService = async () => {
+    try {
+        const calendar = await database.prisma.Service.create({
+            data: {
+                name: 'calendar',
+                description: 'Calendar service',
+                isEnable: true,
+                primaryColor: '#4285F4',
+                secondaryColor: '#FFFFFF',
+                icon: './assets/icons/calendar.png',
```
Comment (ZQUEMA) [inline server/services/calendar/calendar_init.js:16]: Did you add the icon ?

File: server/services/calendar/calendar_init.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+                Actions: {},
+                Reactions: {
+                    create: [
+                        {
+                            name: 'create_event',
+                            code: 'CAL-02',
```
Comment (ZQUEMA) [inline server/services/calendar/calendar_init.js:22]: Why is it '02' and not '01' ?

**Review by ZQUEMA — CHANGES_REQUESTED** [review]
Needed changes

File: server/services/calendar/calendar_init.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                description: 'Calendar service',
+                isEnable: true,
+                primaryColor: '#4285F4',
+                secondaryColor: '#FFFFFF',
+                icon: './assets/icons/calendar.png',
+                Actions: {},
+                Reactions: {
+                    create: [
+                        {
+                            name: 'create_event',
```
Comment (tbellicha) [inline server/services/calendar/calendar_init.js]: Use CamelCase instead

File: server/services/calendar/reactions/create_event.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+      })
+      return response.data
+    }
+  } catch (err) {
+    console.log(`An error occurred: ${err}`)
+    throw err
+  }
+}
+
+const createEventWithAccessToken = async (accessToken, calendarId, summary, description, start, end) => {
```
Comment (tbellicha) [inline server/services/calendar/reactions/create_event.js:93]: Missing docs

File: server/services/calendar/reactions/create_event.js [hunk 5]
```diff
@@ -0,0 +1,107 @@
+const calendar = require('../calendar_init').getCalendarClient()
+const { replaceDynamicParameters } = require('../../glue/dynamic_parameters.js')
+const { google } = require('googleapis')
+
+const calendarCreateEventFromArea = async (Area, dynamicParameters) => {
```
Comment (tbellicha) [inline server/services/calendar/reactions/create_event.js:11]: Missing docs

**Review by tbellicha — CHANGES_REQUESTED** [review]

**Review by ZQUEMA — APPROVED** [review]

**Review by tbellicha — APPROVED** [review]

**Pull request merged**